modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/session/SessionManagerTest.java

```diff
@@ -0,0 +1,128 @@
+package org.apache.ignite.internal.sql.engine.session;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.Map;
+import java.util.UUID;
+import javax.validation.constraints.AssertTrue;
+import org.apache.ignite.internal.sql.engine.property.PropertiesHolder;
+import org.apache.ignite.internal.sql.engine.property.Property;
+import org.apache.ignite.internal.testframework.IgniteTestUtils;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SessionManagerTest {
+
+    private SessionManager sessionMgr;
+    private Map<SessionId, Session> activeSessions;
+
+    @BeforeEach
+    void beforeEach() {
+        sessionMgr = new SessionManager("test", System::currentTimeMillis);
+        activeSessions = IgniteTestUtils.getFieldValue(sessionMgr, "activeSessions");
+    }
+
+    @AfterEach
+    void afterEach() throws Exception {
+        sessionMgr.stop();
+    }
+
+    @Test
+    void createSession() {
+        assertEquals(0, activeSessions.size());
+
+        SessionId sessionId1 = sessionMgr.createSession(1000, null);
+        assertNotNull(sessionId1);
+
+        SessionId sessionId2 = sessionMgr.createSession(1000, null);
+        assertNotNull(sessionId2);
+
+        assertNotEquals(sessionId1, sessionId2);
+        assertEquals(2, activeSessions.size());
+    }
+
+    @Test
+    void sessionGet() {
+        PropertiesHolder propHldr = createPropertyHolder();
+
+        SessionId sessionId = sessionMgr.createSession(12345, propHldr);
+
+        Session session = sessionMgr.session(sessionId);
+        assertNotNull(session);
+        assertSame(propHldr, session.queryProperties());
+        assertEquals(12345, session.getIdleTimeoutMs());
+
+        SessionId unknownSessionId = new SessionId(UUID.randomUUID());
+        assertNull(sessionMgr.session(unknownSessionId));
+    }
+
+    @Test
+    void touchSessionDuringGet() throws InterruptedException {
```

Comment:
It gets tricky when it comes to testing the time related things. That's why I have introduced CurrentTimeProvider interface to be used in SessionManagement. By using this interface you could travel backward and forward in time without wasting time in sleeping.

So please get rid of all `Thread.sleep()` in the class

Reply:
@korlov42 In general I agree with you, however not all cases could be checked without `Thread.sleep()`, a s example we should check that session is prolonging expiration time after touch. WDYT, could we keep this part and rework another one?

Reply:
The mentioned case could be easily covered like this:
```
var clock = new AtomicLong(1L)
var sessionManager = new SessionManager(clock::get)
var session = sessionManager.createSession(5sec)

clock.addAndGet(4) // wait for 4 secs

session.touch()

clock.addAndGet(4) // wait another 4 secs

assertFalse(session.expired())
```

Reply:
please look new code in the patch - I think no simple way don't use Thread sleep for the case